### PR TITLE
Prac 08 feedback

### PR DESCRIPTION
**Pull request by Emma-Naumann**
@Connorw-dev please review this practical, especially general conventions and best practice naming etc.

### CODE REVIEW

File: prac_08/dynamic_labels.py [hunk 1]
```diff
@@ -4,4 +4,32 @@
...
+    def build(self):
+        """Build the Kivy GUI."""
+        self.title = "Dynamic Labels"
+        self.root = Builder.load_file('dynamic_labels.kv')
+        self.create_labels()
+        return self.root
+
+    def create_labels(self):
+        """Create labels from list and add them to the GUI."""
+        for name in self.names:
```
Comment (Connorw-dev) [inline prac_08/dynamic_labels.py:29]: for a simple label like this you could consider list comprehension. Ie. `[self.root.ids.main.add_widget(Label(text=name)) for name in self.names]`

File: prac_08/convert_miles_km.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+Emma Naumann
+"""
+
+from kivy.app import App
+from kivy.lang import Builder
+from kivy.core.window import Window
+
+__author__ = 'Emma Naumann'
+
+MILES_TO_KM = 1.60934
```
Comment (Connorw-dev) [inline prac_08/convert_miles_km.py:13]: This is the same naming convention as a dictionary (KEY_TO_VALUE) but it isn't a dictionary.

File: prac_08/convert_miles_km.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        self.root.ids.input_miles.text = str(value)
+        self.handle_calculate()
+
+    def get_valid_miles(self):
+        """Get valid distance in miles from text input, return 0 if value error."""
+        try:
+            value = float(self.root.ids.input_miles.text)
+            return value
+        except ValueError:
+            return 0
```
Comment (Connorw-dev) [inline prac_08/convert_miles_km.py:42]: `return None` might be nicer if that works

File: prac_08/KivyDemos-master/squaring.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+Kivy GUI program to square a number
+Lindsay Ward, IT@JCU
+Started 13/10/2015
+"""
+
+from kivy.app import App
+from kivy.lang import Builder
+from kivy.core.window import Window
+
+__author__ = 'Lindsay Ward'
```
Comment (Connorw-dev) [inline prac_08/KivyDemos-master/squaring.py:12]: Hi Lindsay!
Should probably change this and the docstring

File: prac_08/convert_miles_km.kv [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+            on_text: app.handle_calculate()
+        BoxLayout:
+            orientation: 'vertical'
+            Button:
+                text: 'Up'
+                on_press: app.handle_increment(1)
+            Button:
+                text: 'Down'
+                on_press: app.handle_increment(-1)
+#    Button:
```
Comment (Connorw-dev) [inline prac_08/convert_miles_km.kv:20]: Remember to remove your "testing" comments in the end if you aren't planning on adding that code back.

**Review by Connorw-dev — COMMENTED** [review]
Some minor errors in there. I didn't test your app GUI.

File: prac_08/dynamic_labels.py (hunk 1 above)
Comment (Emma-Naumann) [reply prac_08/dynamic_labels.py:29]: Good suggestion thanks :)

**Pull request merged**